packages/relay/src/formatters.ts

```diff
@@ -291,14 +291,30 @@ const isHex = (value: string): boolean => {
   return hexRegex.test(value);
 };
 
-// Returns the sum of all transfer amounts for the specified account. The amount is negative if the account is charged,
-// it is positive if the account is receiving it, thus the amount is first negated and then added to the sum.
+/**
+ * Returns the sum of all transfer amounts for the specified account. The amount is negative if the account is charged,
+ * it is positive if the account is receiving it, thus the amount is first negated and then added to the sum.
+ * @param {TransactionRecord} transactionRecord - The record of the transaction including transfers.
+ * @param {string} accountId - The ID of the account for which the transfer amounts are being calculated.
+ * @returns {number} - The sum of transfer amounts for the specified account.
+ * @throws {Error} - Throws an error if the `transactionRecord` is invalid or if the `accountId` is not found.
+ */
 const getTransferAmountSumForAccount = (transactionRecord, accountId: string): number => {
-  return transactionRecord.transfers
-    .filter((transfer) => transfer.accountId.toString() === accountId)
-    .reduce((acc, transfer) => {
-      return BN.sum(acc, transfer.amount.toTinybars().negate()).toNumber();
-    }, 0);
+  const isFromConsensusNode = process.env.GET_RECORD_DEFAULT_TO_CONSENSUS_NODE === 'true';
+
+  if (isFromConsensusNode) {
+    return transactionRecord.transfers
+      .filter((transfer) => transfer.accountId.toString() === accountId)
+      .reduce((acc, transfer) => {
+        return BN.sum(acc, transfer.amount.toTinybars().negate()).toNumber();
```

Comment:
Good point! I believe this code was written by @Ivo-Yankov, and I think `transfer.amount.toTinybars()` returns a BN, while `acc` does not. Therefore, `acc` needs to be cast. I'll test to see if it's possible to avoid casting them into BN and then back into a number.

Reply:
You can try this:
```suggestion
        return acc - transfer.amount.toTinybars().toNumber();
```

Reply:
Thanks! Yes that's what I had for the new updated solution!